### DISCUSSION TITLE
N-terminal acetylation (NTA) for peptide chain

### DISCUSSION THREAD

**Discussion by shyamramesh** [rdkit/rdkit#7171] [Q&A]
is there a straight forward way to do this in RDKit? 
or
identify the n-terminus
remove hydrogen 
add the acetyl group
are these steps to do it?
any code snippet will help

**Comment by ptosco** [comment]
@shyamramesh You may do this using a chemical reaction:
```
from rdkit import Chem
from rdkit.Chem import rdChemReactions

rxn = rdChemReactions.ReactionFromSmarts("[$([NX3H2,$([NX3H1](-[CX4])-[!$([CX3]=[OX1])])]-[CX4]-[CX3]=[OX1]):1]>>[N:1]-C(=O)-C")
val_lys_ala = Chem.MolFromSmiles("C[C@@H](C(=O)O)NC(=O)[C@H](CCCCN)NC(=O)[C@H](C(C)C)N")
val_lys_ala
```
![image](https://github.com/rdkit/rdkit/assets/5244385/cd3e621d-a10f-4268-82a7-28d7fbd1a595)
```
products = rxn.RunReactants((val_lys_ala,))
products
((<rdkit.Chem.rdchem.Mol at 0x7ff2f26bacf0>,),)
Chem.SanitizeMol(products[0][0])
products[0][0]
```
![image](https://github.com/rdkit/rdkit/assets/5244385/683a7fef-987a-465c-8fda-c93614181be7)
